src/object/schema_manager.c

```diff
@@ -2249,23 +2277,88 @@ sm_qualifier_name (const char *name, char *buf, int buf_size)
       return NULL;
     }
 
+  assert (buf != NULL);
+  assert (buf_size > 0);
+
   /* If the name is not a user-specified name, NULL is returned. */
   dot = strchr (name, '.');
   if (dot == NULL)
     {
       return NULL;
     }
 
-  len = STATIC_CAST (int, dot - name);
+  name_size = STATIC_CAST (int, dot - name);
+  if (name_size >= buf_size || name_size >= DB_MAX_USER_LENGTH)
+    {
+      /* exit on error */
 
-  assert (buf != NULL);
-  assert (len > 0 && len < buf_size);
-  memcpy (buf, name, len);
-  buf[len] = '\0';
+      char name_copy[DB_MAX_USER_LENGTH] = { '\0' };
+      int n_bytes = (buf_size <= DB_MAX_USER_LENGTH) ? buf_size : DB_MAX_USER_LENGTH;
+
+      if (name_size < DB_MAX_USER_LENGTH)
+	{
+	  memcpy (name_copy, name, name_size);
+	  name_copy[name_size] = '\0';
+	}
+      else
+	{
+	  memcpy (name_copy, name, (DB_MAX_USER_LENGTH - 4));
+	  memcpy (name_copy, "...", 4);
```

Comment:
it's a proper copy operation ?  maybe, name_copy has only string "..."

Reply:
Changed. Thank you.